docs/architecture/joins.txt

```diff
@@ -119,13 +127,13 @@ loop to acquire the data before being able to join. After finding the locations
 of the required shards (which is done in the planning stage), the smaller data
 set (based on the row count) is broadcast amongst all the nodes holding the
 shards they are joined with. After that, each of the receiving nodes can start
-running a nested loop on the subset it has just received. Finally, the results
-are pushed to the original (planner) node to merge and return the results to
-the requesting client (see
-:ref:`joins_figure_2`).
+running a nested loop on the subset it has just received. Finally, these
+intermediate results are pushed to the original (planner) node to merge and
```

Comment:
shall we say "handler" node instead of the "planner" node? I think we generally use the "handler" term no?